dask/array/core.py

```diff
@@ -1938,14 +1998,112 @@ def normalize_chunks(chunks, shape=None):
             raise ValueError("Input array has %d dimensions but the supplied "
                              "chunks has only %d dimensions" %
                              (len(shape), len(chunks)))
-        if not all(c == s or (math.isnan(c) and math.isnan(s))
+        if not all(c == s or (math.isnan(c) or math.isnan(s))
                    for c, s in zip(map(sum, chunks), shape)):
             raise ValueError("Chunks do not add up to shape. "
                              "Got chunks=%s, shape=%s" % (chunks, shape))
 
     return tuple(tuple(int(x) if not math.isnan(x) else x for x in c) for c in chunks)
 
 
+def auto_chunks(chunks, shape, limit, dtype, previous_chunks=None):
+    """ Determine automatic chunks
```

Comment:
Would be good to have a descriptor (maybe in more detail) about what "auto" means here. What is the gist of it, how do the parameters factor in, etc.

Reply:
I've added a modest description of the behavior.